chain/types/actor_event.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type ActorEventBlock struct {
+	// what value codec does client want to match on ?
+	Codec uint64 `json:"codec"`
+	// data associated with the "event key"
+	Value []byte `json:"value"`
+}
+
+type SubActorEventFilter struct {
+	Filter  ActorEventFilter `json:"filter"`
+	Prefill bool             `json:"prefill"`
```

Comment:
If client is subscribing to events using the `SubscribeActorEvents(ctx context.Context, filter *types.SubActorEventFilter) (<-chan *types.ActorEvent, error)
}` API, they can use the `Prefill` flag here to recieve pre-existing events that match the filter before recieving new/real-time events.

Reply:
wouldn't that be implied, and controlled by the min-epoch in the filter though? or do we want to be doubly sure and have a default `-f` behaviour so we're not being wasteful?

Reply:
I think we want to ensure clients explicitly ask for historical events so as to not be wasteful. 

A min-epoch of -1 could still mean that "I don't want historical events but I want everything starting from now" so best to have a `Prefill` flag.

Reply:
worth documenting this since you needed to explain it in the comments